x/auth/keeper/keeper.go

```diff
@@ -277,3 +279,41 @@ func (ak AccountKeeper) GetParams(ctx context.Context) (params types.Params) {
 	}
 	return params
 }
+
+// -------------------------------------
+// Unordered Sequence management methods
+// -------------------------------------
+
+// ContainsUnorderedSequence reports whether the sender has used this timestamp already.
+func (ak AccountKeeper) ContainsUnorderedSequence(ctx sdk.Context, sender []byte, timestamp time.Time) (bool, error) {
+	return ak.UnorderedSequences.Has(ctx, collections.Join(uint64(timestamp.UnixNano()), sender))
```

Comment:
Why do we need to cast to `uint64`, can we just use `int64` as the collections type?

Also FWIW `UnixNano` will cause this code to break after 2262 😜

Reply:
oops, switched to int64 keys